dev/Installer/src/install.cpp

```diff
@@ -0,0 +1,172 @@
+#include "pch.h"
+#include "packages.h"
+#include "install.h"
+
+using namespace winrt;
+using namespace Windows::ApplicationModel;
+using namespace Windows::Foundation;
+using namespace Windows::Management::Deployment;
+using namespace Windows::System;
+
+winrt::hresult add_package(Uri packageUri)
+{
+    PackageManager packageManager;
+    auto deploymentOperation{ packageManager.AddPackageAsync(packageUri, nullptr, DeploymentOptions::None) };
+    deploymentOperation.get();
+    if (deploymentOperation.Status() != AsyncStatus::Completed)
+    {
+        return deploymentOperation.ErrorCode();
+    }
+
+    return S_OK;
+}
+
+winrt::hresult provision_package(std::wstring packageFamilyName)
+{
+    PackageManager packageManager;
+    auto deploymentOperation{ packageManager.ProvisionPackageForAllUsersAsync(packageFamilyName.c_str()) };
+    deploymentOperation.get();
+
+    if (deploymentOperation.Status() != AsyncStatus::Completed)
+    {
+        return deploymentOperation.ErrorCode();
+    }
+
+    return S_OK;
+}
+
+bool is_architecture_applicable(ProcessorArchitecture arch)
+{
+    SYSTEM_INFO systemInfo;
+    GetNativeSystemInfo(&systemInfo);
+    ProcessorArchitecture systemArch = static_cast<ProcessorArchitecture>(systemInfo.wProcessorArchitecture);
+
+    if (arch == systemArch)
+    {
+        return true;
+    }
+
+    // On x64 systems, x86 architecture is also applicable
+    if (systemArch == ProcessorArchitecture::X64 && arch == ProcessorArchitecture::X86)
+    {
+        return true;
+    }
+
```

Comment:
Missing ARM64. Also be careful, `GetNativeSystemInfo` will return `PROCESSOR_ARCHITECTURE_INTEL` on that hardware with native executables.

Reply:
@DrusTheAxe arm64 images return PROCESSOR_ARCHITECTURE_INTEL. (Yeah, I know. That platform is slightly cursed.)

Reply:
> Missing ARM64. Also be careful, `GetNativeSystemInfo` will return `PROCESSOR_ARCHITECTURE_INTEL` on that hardware with native executables.

What is the correct way to test that the native system is in fact arm64 and not x86?

Reply:
I'd replace `GetNativesystemInfo` usage here with [`IsWow64Process2`](https://docs.microsoft.com/en-us/windows/win32/api/wow64apiset/nf-wow64apiset-iswow64process2). Then you can inspect `pNativeMachine`, which returns the correct `IMAGE_FILE_MACHINE_ARM64`.

Reply:
I added Arm64 and what should be Arm64 detection, but I noticed that visual studio's drop-down menu architecture configuration did not detect _M_ARM64 and instead appears to be _M_IX86. This is a more complicated issue to resolve since it has to do with both packaging the correct frameworks in the Arm64 version as well as runtime detection to solve it, so I created a new issue to track it - #479

I suspect resolving that is going to be a non-trivial amount of work and should not block this initial installer deployment.